cmd/zoekt-sourcegraph-indexserver/queue.go

```diff
@@ -124,7 +129,112 @@ func (q *Queue) Iterate(f func(*IndexOptions)) {
 	q.mu.Lock()
 	defer q.mu.Unlock()
 	for _, item := range q.items {
-		f(&item.opts)
+		f(&item.Opts)
+	}
+}
+
+// debugEncodeSortedGobStream gob-encodes all items contained in the queue
+// (sorted by priority) to the provided writer.
+func (q *Queue) debugEncodeSortedGobStream(w io.Writer) error {
+	enc := gob.NewEncoder(w)
+
+	var err error
+	q.debugIteratedOrdered(func(item *queueItem) {
+		if err != nil {
+			return
+		}
+
+		err = enc.Encode(streamQueueReply{
+			Type: streamReplyQueueItem,
+			Item: item,
+		})
+	})
+
+	if err != nil {
+		reply := streamQueueReply{
+			Type:        streamReplyError,
+			ErrorString: fmt.Sprintf("encoding queueItem: %s", err),
+		}
+
+		replyErr := enc.Encode(reply)
+		if replyErr != nil {
+			return fmt.Errorf("encoding streamError reply: %w", replyErr)
+		}
+
+		return fmt.Errorf("encoding queueItem: %w", err)
+	}
+
+	err = enc.Encode(streamQueueReply{
+		Type: streamReplyDone,
+	})
+
+	if err != nil {
+		return fmt.Errorf("terminating stream: %w", err)
+	}
+
+	return nil
+}
+
+// debugIteratedOrdered will call f on each queueItem (sorted by indexing priority)
+// known to the queue, including queueItems that have been popped from the queue).
+//
+// Note: The mutex is held during all calls to f. Callers must not modify the queueItems.
+func (q *Queue) debugIteratedOrdered(f func(*queueItem)) {
```

Comment:
The current CLI  output doesn't use any fields that are unique to `*queueItem`, but my next follow up after this lands is to add a `firstInsertedIntoQueue time.Time` field to every queueItem so that it can be displayed in the output. This is why I added a new kind of iteration method to the queue struct. 

I don't think it makes sense to add this new field to IndexOptions since it's essentially metadata about the queue iteslf.